test/integration/replicaset/replicaset_test.go

```diff
@@ -930,3 +931,166 @@ func TestFullyLabeledReplicas(t *testing.T) {
+	}
+
+	updateRS(t, rsClient, rs.Name, func(rs *v1beta1.ReplicaSet) {
+		var finalizers []string
+		// remove fakeFinalizer
+		for _, finalizer := range rs.Finalizers {
+			if finalizer != fakeFinalizer {
+				finalizers = append(finalizers, finalizer)
+			}
+		}
+		rs.Finalizers = finalizers
+	})
+
+	rsClient.Delete(rs.Name, nil)
+}
+
+// TODO(dixudx): enable this test when replicaset controller is bumped to apps/v1
+//func TestReplicaSetsAppsV1DefaultGCPolicy(t *testing.T) {
+//	s, closeFn, rm, informers, c := rmSetup(t)
+//	defer closeFn()
+//	ns := framework.CreateTestingNamespace("test-default-gc-appsv1", s, t)
+//	defer framework.DeleteTestingNamespace(ns, s, t)
+//	stopCh := runControllerAndInformers(t, rm, informers, 0)
+//	defer close(stopCh)
+//
+//	rsName := "rs"
+//	fakeFinalizer := "kube.io/dummy-finalizer"
+//	replicasCopy := int32(2)
+//
+//	rs := &appsv1.ReplicaSet{
+//		TypeMeta: metav1.TypeMeta{
+//			Kind:       "ReplicaSet",
+//			APIVersion: "apps/v1",
+//		},
+//		ObjectMeta: metav1.ObjectMeta{
+//			Namespace:    ns.Name,
+//			GenerateName: "gc-rs-test-appsv1",
+//			Name:         rsName,
+//			Finalizers:   []string{fakeFinalizer},
+//		},
+//		Spec: appsv1.ReplicaSetSpec{
+//			Replicas: &replicasCopy,
+//			Selector: &metav1.LabelSelector{
+//				MatchLabels: labelMap(),
+//			},
+//			Template: v1.PodTemplateSpec{
+//				ObjectMeta: metav1.ObjectMeta{
+//					Labels: labelMap(),
+//				},
+//				Spec: v1.PodSpec{
+//					Containers: []v1.Container{
+//						{
+//							Name:  "fake-name",
+//							Image: "fakeimage",
+//						},
+//					},
+//				},
+//			},
+//		},
+//	}
+//
+//	rsClient := c.AppsV1().ReplicaSets(rs.Namespace)
+//	if _, err := rsClient.Create(rs); err != nil {
+//		t.Fatalf("Failed to create replica set %s: %v", rs.Name, err)
+//	}
+//
+//	// Verify .Status.Replicas is equal to .Spec.Replicas
+//	if err := wait.PollImmediate(interval, timeout, func() (bool, error) {
+//		newRS, err := rsClient.Get(rs.Name, metav1.GetOptions{})
+//		if err != nil {
+//			return false, err
+//		}
+//		return newRS.Status.Replicas == *rs.Spec.Replicas, nil
+//	}); err != nil {
+//		t.Fatalf("Failed to verify .Status.Replicas is equal to .Spec.Replicas for rs %s: %v", rs.Name, err)
+//	}
+//
+//	// Verify RS creates 2 pods
+//	podClient := c.CoreV1().Pods(ns.Name)
+//	pods := getPods(t, podClient, labelMap())
+//	if len(pods.Items) != 2 {
+//		t.Fatalf("len(pods) = %d, want 2", len(pods.Items))
+//	}
+//
+//	err := rsClient.Delete(rs.Name, getOrphanOptions())
+//	if err != nil {
+//		t.Fatalf("Failed to delete rs: %v", err)
+//	}
+//
+//	// Verify no new finalizer has been added
+//	if err := wait.PollImmediate(interval, timeout, func() (bool, error) {
+//		newRS, err := rsClient.Get(rs.Name, metav1.GetOptions{})
+//		if err != nil {
+//			return false, err
+//		}
+//		t.Logf("getting the fianlizer: %s", newRS.Finalizers)
+//		return reflect.DeepEqual(newRS.Finalizers, []string{fakeFinalizer}), nil
```

Comment:
Before checking that no finalizer was added, make sure that `DeletionTimestamp != nil`. If it's `nil`, keep polling.